plugin/pkg/up/up_test.go

```diff
@@ -38,3 +39,35 @@ func TestUp(t *testing.T) {
 		t.Errorf("Expected hits to be %d, got %d", 1, h)
 	}
 }
+
+func TestDoBackoff(t *testing.T) {
+	pr := New()
+	wg := sync.WaitGroup{}
+	pr.Start(time.Millisecond)
+	pr.Lock()
+	pr.expBackoff.RandomizationFactor = 0
+	pr.expBackoff.Multiplier = 2
+	pr.Unlock()
+	defer pr.Stop()
+
+	i := 0
+	upfunc := func() error {
+		if i > 5 {
+			wg.Done()
+			return nil
+		}
+		i++
+
+		return errors.New("test")
+	}
+	start := time.Now()
+	wg.Add(1)
+	pr.Do(upfunc)
+	wg.Wait()
+
+	elapsed := time.Now().Sub(start)
+	expected := time.Millisecond * (1 + 2 + 4 + 5 + 6) // plus execution time
```

Comment:
time.Millisecond * (1 + 2 + 4 + 8 +16)

Reply:
> this is why the hc is 0.5s (fixed). With exp backoff you also need random probes to see if upstream is healthy again, that's not done.
> Please explain why 2 HC checks per second are problematic, this is a tiny amount of traffic and you want to react fast when an upstream is healthy again.

@miekg, if the current behavior is intentional and acceptable to you, I can close this PR.

Reply:
@miekg, Ah - are you saying you would rather remove the existing HC backoff behavior?

Reply:
Was the intent of #3970 to revert the incremental backoff code added in #3643?